Tasks/SonarQubePostTest/SonarQubeMetrics.ps1

```diff
@@ -38,18 +38,38 @@ function WaitForAnalysisToFinish
     Assert (![String]::IsNullOrEmpty($analysisId)) "Could not fetch the analysis id"
 }
 
+
+function FetchQualityGateDetails
+{
+    param ([Array]$errors,[Array]$warnings)
+    
+    $analysisId = GetTaskContextVariable "MSBuild.SonarQube.AnalysisId"             
+    Assert (![String]::IsNullOrEmpty($analysisId)) "WaitForAnalysisToFinish should be called first."
+       
+    $response = InvokeGetRestMethod "/api/qualitygates/project_status?analysisId=$analysisId" $true
```

Comment:
Good catch! Actually calling api/qualitygates/project_status using the analysisId parameter is supported since 5.3 whereas projectId  / projectKey have been introduced in 5.4. I have used api/qualitygates/project_status with analysisId for the build breaker and this was tested against SonarQube 5.3;